### ISSUE TITLE
Facebook login system (if anyone has it, please share it!)

### ISSUE THREAD

**Issue by CalvinDKL** [CigarProject/Cigar#446]
Hello! So this would be a cool thing, if anyone has the code for it.
Please share it!

Thank you

**Comment by HAlexTM** [comment]
This requires API keys from the Facebook Developers website, some code will not help you that much

**Comment by CalvinDKL** [comment]
http://www.oauthlogin.com/#Architecture

This seems promising

**Comment by HAlexTM** [comment]
That's why you require hundreds lines of code

**Comment by CalvinDKL** [comment]
Check the link, its a full login system.

**Comment by HAlexTM** [comment]
That you need to integrate with your client & server mhm...

**Comment by CalvinDKL** [comment]
well its easier than making my own login system

**Comment by CalvinDKL** [comment]
I just dont know how to do, so certain fb ids have certain things in the server.

**Comment by Luka967** [comment]
@CalvinDKL Login system requires heavy HTML & JS code editing and a SQL server for unique user database (Since you want Facebook logins, requires API keys just like HAlexTM mentioned). And believe me, making your own login system IS a huge lot easier, better and safer. Best made by using SQL protocols back and forth and PHP files, it would be very versatile. You don't need auth tokens like agar.io has or session queries the site mentions (Session tokens are just for preserving login info).

However, certain things in the server for every ID is tricky as this adds new packets and requires a custom backend server (the Ogar server) for handling the packets. It would add another level of complexity.

If you really want both, make the login system first. Scale it by adding session tokens, fix bugs and security holes. Then, and only then, edit the client-server protocol for Ogar.

**Comment by HAlexTM** [comment]
Will re-open if @Megabyte918 will do something related to this server-side.

**Comment by MastaCoder** [comment]
@HAlexTM @Luka967 I made a project a long time back called MyUsers, it was pretty much an advanced version of UserCake. I managed to implement this system with my website, but the whole skin case thing required hundreds of lines to get it to work. (Took me about 2 weeks of straight working in the summer)

Login system isn't hard to make with MyUsers. It's just finding a server to add support to that is hard. We can't really make our own server either because MultiOgar-Edited is being updated constantly.
